### DISCUSSION TITLE
Get the localized pathname that the user sees in the browser?

### DISCUSSION THREAD

**Discussion by Pustinyak** [vercel/next.js#19077] [Help] [answered]
Using the new i18n routing, the locale is stripped from the pathname when I try to get it.

In the documentation for `getInitialProps` is described a way to get the url using `asPath`:
> `asPath` - String of the actual path (including the query) **shown in the browser**

However that's not true with the new i18n pathname routing. If the url is "**/en/something**" then `asPath` points to "**/something**".

I've tried using `resolvedUrl` and `req.url` from `getServerSideProps`'s context but there's no difference.

### Edit:
I found the issue here - https://github.com/vercel/next.js/issues/18782

Is there a way to get the localized pathname or somehow to know if the user came from a localized path?

For example: If the default locale is EN, there's no way to check if the user came from "**/en/something**" or "**/something**".

**Comment by Pustinyak** [comment, marked as answer]
After a lot of debugging I found a good way to do a check for this:

Use `context.req.__nextStrippedLocale`. It works on `getServerSideProps` and `getInitialProps` and it returns `true` if the current pathname is localized, otherwise it's `undefined` if there is no locale in the pathname.

**Comment by goodyInt** [comment]
thanks for this solution, @EmilTruthful - I had this problem on a big international localization project and using work around I am able to enable the default locale from all locales without it rerouting to the host country. Thanks again for posting it!
